tests/client/test_machine.py

```diff
@@ -0,0 +1,58 @@
+import numpy as np
+import tinychain as tc
+import unittest
+from sklearn.model_selection import train_test_split
+from testutils import ClientTest
+
+Dense = tc.tensor.Dense
+ENDPOINT = "/transact/hypothetical"
+LEARNING_RATE = tc.F32(0.01)
+MAX_ITERATIONS = 10
+NUM_EXAMPLES = 20
+
+
+class Logistic_Regression():
+    def fit (self,inputs ,labels):
+        layer0= tc.ml.nn.DNNLayer.create('layer0', inputs.shape[1], 1, tc.ml.Sigmoid())
+        logistic = tc.ml.nn.Sequential.load([layer0])
+        def train_while(i: tc.UInt, output: tc.tensor.Dense):
+            return (i <= MAX_ITERATIONS).logical_and(((output > 0.5) != labels).any())
+        
+        def cost(self,output, labels, dl=False):
+            if dl:
+                return output.sub(labels).mul(2).mean()
+            return output.sub(labels).pow(2).mean()        
+        
+        # for trial only , Adam must be replaced by Gradient Descent !
+        optimizer = tc.ml.optimizer.Adam.create(param_list=logistic.get_param_list(), lr=LEARNING_RATE)
+        result= tc.ml.optimizer.train(logistic, optimizer, inputs, labels, cost, train_while)
```

Comment:
This doesn't work because `train` contains a `closure` that tries to capture states from the outer context, but there is no outer context here. I think you can fix this problem by making `fit` a `tc.post_op`.